packages/backend/src/services/DashboardService/DashboardService.ts

```diff
@@ -169,6 +170,13 @@ export class DashboardService {
             throw new ForbiddenError();
         }
         if (
+            user.ability.cannot(
+                'manage',
+                subject('Project', {
+                    organizationUuid: dashboard.organizationUuid,
+                    projectUuid: dashboard.projectUuid,
+                }),
+            ) &&
```

Comment:
Since the admins now have full access to private spaces I think we should have this validation in the `hasSpaceAccess` util function. 90% of the places we use this function need the new check. 
The only exceptions are during listing all spaces, all dashboards, all charts. 

Example:
```
export const hasSpaceAccess = (
    space: Pick<Space, 'isPrivate' | 'access'>,
    user: SessionUser,
    checkAdminAccess: boolean = true,
): boolean => {
const hasAdminAccess = user.ability.can(
                'manage',
                subject('Project', {
                    organizationUuid: space.organizationUuid,
                    projectUuid: space.projectUuid,
                }),
            )

    const hasAccess =
        !space.isPrivate ||
        space.access.find(
            (userAccess) =>
                userAccess.userUuid === userUuid && userAccess.role !== null,
        ) !== undefined;

    return checkAdminAccess ? hasAdminAccess || hasAccess : hasAccess;
};
```

Reply:
Yup, that makes sense, I'll change it